x-pack/solutions/chat/plugins/wci-index-source/public/integration/configuration.tsx

```diff
@@ -50,6 +58,35 @@ export const IndexSourceConfigurationForm: React.FC<IntegrationConfigurationForm
   ];
 
   const { generateSchema } = useGenerateSchema();
+  const { isLoading, autocompleteQuery } = useIndexNameAutocomplete();
```

Comment:
I would have personally integrated the "debounce" effect in the `useIndexNameAutocomplete` hook, as I assume this is part of the hook's behavior and something we want to always have.

Something like

```ts
// component
const [query, setQuery] = useState('');
const { isLoading, indexList } = useIndexNameAutocomplete({ query });

// hook

useIndexNameAutocomplete({ query }) {

  const [debouncedQuery, setDebounceQuery] = useState<string>(query);
  useDebounce(() => {
     setDebounceQuery(query);
  }, 250, [query]);

  const { isLoading, data } = useQuery({
    queryKey: ['index-name-autocomplete', debouncedQuery],
    queryFn:
      () =>
      async () => {
        return http.pseudocode(url, {body: { query }})
      },
      initialData: () => [],
  });
}